src/test/scala/is/hail/distributedmatrix/BlockMatrixSuite.scala

```diff
@@ -36,47 +37,47 @@ class BlockMatrixSuite extends SparkSuite {
-
-  def blockMatrixPreGen(blockSize: Int, transposed: Boolean): Gen[BlockMatrix] = for {
-    (l, w) <- Gen.nonEmptySquareOfAreaAtMostSize
-    m <- blockMatrixPreGen(l, w, blockSize, transposed)
-  } yield m
-
-  def blockMatrixPreGen(rows: Int, columns: Int, blockSize: Int): Gen[BlockMatrix] =
-    Gen.coin().flatMap(blockMatrixPreGen(rows, columns, blockSize, _))
-
-  def blockMatrixPreGen(rows: Int, columns: Int, blockSize: Int, transposed: Boolean): Gen[BlockMatrix] = for {
-    arrays <- Gen.buildableOfN[Seq, Array[Double]](rows, Gen.buildableOfN(columns, arbDouble.arbitrary))
+  private val defaultBlockSize = choose(0, 1 << 15)
+  private val defaultDims = nonEmptySquareOfAreaAtMostSize
+  private val defaultTransposed = coin()
+  private val defaultElement = arbitrary[Double]
+  def blockMatrixGen(
+    blockSize: Gen[Int] = defaultBlockSize,
+    dims: Gen[(Int, Int)] = defaultDims,
+    transposed: Gen[Boolean] = defaultTransposed,
+    element: Gen[Double] = defaultElement
+  ): Gen[BlockMatrix] = for {
+    blockSize <- blockSize
+    (rows, columns) <- dims
+    transposed <- transposed
+    arrays <- buildableOfN[Seq, Array[Double]](rows, buildableOfN(columns, element))
     m = toBM(arrays, blockSize)
   } yield if (transposed) m.t else m
 
-  val squareBlockMatrixGen = for {
-    size <- Gen.size
-    l <- Gen.interestingPosInt
-    s = math.sqrt(math.min(l, size)).toInt
-    preBlockSize <- Gen.interestingPosInt
-    blockSize = math.sqrt(preBlockSize).toInt
-    m <- blockMatrixPreGen(s, s, blockSize)
-  } yield m
-
-  val blockMatrixGen = for {
-    blockSize <- Gen.interestingPosInt
-    m <- blockMatrixPreGen(blockSize)
-  } yield m
-
-  val twoMultipliableBlockMatrices = for {
-    Array(rows, inner, columns) <- Gen.nonEmptyNCubeOfVolumeAtMostSize(3)
-    preBlockSize <- Gen.interestingPosInt
-    blockSize = math.pow(preBlockSize, 1.0/3.0).toInt
-    transposed <- Gen.coin()
-    l <- blockMatrixPreGen(rows, inner, blockSize, transposed)
-    r <- blockMatrixPreGen(inner, columns, blockSize, transposed)
+  def squareBlockMatrixGen(
+    transposed: Gen[Boolean] = defaultTransposed,
+    element: Gen[Double] = defaultElement
+  ): Gen[BlockMatrix] = blockMatrixGen(
+    blockSize = interestingPosInt.map(math.sqrt(_).toInt),
+    dims = for {
+      size <- size
+      l <- interestingPosInt
+      s = math.sqrt(math.min(l, size)).toInt
+    } yield (s, s),
+    transposed = transposed,
+    element = element
+  )
+
+  def twoMultipliableBlockMatrices(element: Gen[Double] = arbitrary[Double]): Gen[(BlockMatrix, BlockMatrix)] = for {
+    Array(rows, inner, cols) <- nonEmptyNCubeOfVolumeAtMostSize(3)
+    blockSize <- interestingPosInt.map(math.pow(_, 1.0/3.0).toInt)
+    transposed <- coin()
+    l <- blockMatrixGen(const(blockSize), const(rows -> inner), const(transposed), element)
+    r <- blockMatrixGen(const(blockSize), const(inner -> cols), const(transposed), element)
```

Comment:
Do you need to specify `transposed` here? It's the same as `defaultTransposed` above. Or was it to avoid writing out the argument name for all of the other parameters?